demo/chat_pdf_streamlit_ui.py

```diff
@@ -31,132 +31,96 @@
 Notice the -- to separate our program's cmdline args from streamlit's
 streamlit run demo/chat_pdf_streamlit_ui.py -- --host=http://my-llm-host --port=8000
 '''
-
 import asyncio
+import os
 
-# Could choose to use controls in streamlit rather than cmdline
-import click
+from dotenv import load_dotenv
 
 import streamlit as st
 from PyPDF2 import PdfReader
 
-import langchain
-from langchain.text_splitter import CharacterTextSplitter
-from langchain.chains.question_answering import load_qa_chain
-from langchain.vectorstores import Qdrant
-from langchain.embeddings import SentenceTransformerEmbeddings
-from langchain.callbacks.manager import CallbackManager
-from langchain.callbacks.streaming_stdout import StreamingStdOutCallbackHandler
-from langchain import OpenAI  # Using the API, though on a self-hosted LLM
-
-from ogbujipt.async_helper import schedule_llm_call
-from ogbujipt.config import openai_emulation
+from sentence_transformers import SentenceTransformer
 
-# Workaround for AttributeError: module 'langchain' has no attribute 'verbose'
-# see: https://github.com/hwchase17/langchain/issues/4164
-langchain.verbose = False
-
-QA_CHAIN_TYPE = 'stuff'
-PDF_USER_QUESTION_PROMPT = 'Ask a question about your PDF:'
-# LLM max context size
-# FIXME: Probably want to get this value from the remote API. Figure out how
-N_CTX = 2048
+from qdrant_client import QdrantClient
+from qdrant_client.http import models
 
-# FIXME: Parameterize these, perhaps in streamlit controls, also how many k docs to return
+from ogbujipt.async_helper import schedule_llm_call
+from ogbujipt.config import openai_emulation, openai_live, HOST_DEFAULT
+from ogbujipt.prompting.basic import context_build, pdelim
+from ogbujipt.text_helper import text_splitter
+from ogbujipt.embedding_helper import initialize_embedding_db, upsert_embedding_db
+
+load_dotenv()  # From .env file
+OPENAI = bool(os.getenv('OPENAI')) or False
```

Comment:
Tricky logic, and not right as is.

```py
OPENAI = bool(os.getenv('OPENAI')) or False
```

For example, `bool('x') → True`, so in effect, any non-empty value will give true, even "False" or "false" or "no". Changing to (probably overly-careful code, but a good demo):

```py
OPENAI = os.getenv('OPENAI', 'False') in ['True', 'true', 'TRUE', 'Yes', 'yes', 'YES']
```

Tip: use `default` param to `os.getenv`, including for subsequent lines.